x-pack/plugin/runtime-fields/src/test/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptFieldTypeTestCase.java

```diff
@@ -301,8 +304,11 @@ protected final void checkExpensiveQuery(BiConsumer<MappedFieldType, SearchExecu
         );
     }
 
-    protected final void checkLoop(BiConsumer<MappedFieldType, SearchExecutionContext> queryBuilder) {
-        Exception e = expectThrows(IllegalArgumentException.class, () -> queryBuilder.accept(loopFieldType(), mockContext()));
+    protected final void checkLoop(BiFunction<MappedFieldType, SearchExecutionContext, Query> queryBuilder) throws IOException {
+        Query query = queryBuilder.apply(loopFieldType(), mockContext(true, loopFieldType()));
```

Comment:
This wasn't really checking an actual production code path before.  We now run a search, which forces the script to look up values triggering the loop handling.